### PR TITLE
feat!: update some CLI arguments / env variables for consistency

### PR DESCRIPTION
**Pull request by mikix**
### Description
- Rename `--study_dir` to `--study-dir`
- Drop the S3 bucket gathering entirely, we can get it from workgroup
- Rename `CUMULUS_LIBRARY_SCHEMA` to `CUMULUS_LIBRARY_DATABASE`
- Mention `CUMULUS_LIBRARY_WORKGROUP` in the docs

I could add some safety fallbacks for old env names, if you like the renames, but feel they are too jarring / sudden.

### Checklist
- [x] Consider if documentation (like in `docs/`) needs to be updated
- [x] Consider if tests should be added
- [x] Run pylint if you're making changes beyond adding studies

### CODE REVIEW

File: cumulus_library/cli.py [hunk 1]
```diff
@@ -15,23 +15,26 @@
...
+# but it is used manually as a quick & dirty alternative to the CLI.
 class CumulusEnv:  # pylint: disable=too-few-public-methods
     """
     Wrapper for Cumulus Environment vars.
     Simplifies connections to StudyBuilder without requiring CLI parsing.
     """
 
     def __init__(self):
-        self.bucket = os.environ.get("CUMULUS_LIBRARY_S3")
+        self.s3_output = os.environ.get("CUMULUS_LIBRARY_S3_OUTPUT")
```
Comment (dogversioning) [inline cumulus_library/cli.py:28]: how would you feel about calling this `s3_staging` instead?

the docs for pyathena leave something to be desired in this regard - but, assuming you're testing in HMS AWS, it might be a bucket permissioning issue on the default athena cache location??

**Review by dogversioning — APPROVED** [review]

File: cumulus_library/cli.py [hunk 2]
```diff
@@ -15,23 +15,26 @@
...
         self.region = os.environ.get("CUMULUS_LIBRARY_REGION", "us-east-1")
         self.workgroup = os.environ.get("CUMULUS_LIBRARY_WORKGROUP", "cumulus")
         self.profile = os.environ.get("CUMULUS_LIBRARY_PROFILE")
-        self.schema_name = os.environ.get("CUMULUS_LIBRARY_SCHEMA")
+        self.database = os.environ.get("CUMULUS_LIBRARY_DATABASE")
```
[the remark this answers is not in the document]
Comment (mikix) [reply cumulus_library/cli.py:32]: That might be true at a technical level. And if you want to keep `schema_name` internally, I'm down with that.

But:
- Obviously, schema is a very overloaded term. Databases hold tables, which hold schemas. It's very weird to talk about databases as schemas themselves.
- In the user experience, they are never referred to as schemas, but always databases. Load up the Athena or Glue web consoles.

So for the user-visible parts (env var and command line arg), I'd strongly push for `database`.

File: cumulus_library/cli.py (hunk 1 above)
Comment (mikix) [reply cumulus_library/cli.py:28]: The docs for `pyathena` are terrible. I've been looking at code for most of my questions, and even then it's not super helpful.

tl;dr; sure `staging` is fine.

But I have some thoughts:
- What does `pyathena` do with this folder? Is it really just the "Query result location" for the workgroup? Like, is it dropping query results there, via Athena queries? Or some other unnamed artifacts...?
- The user would be familiar with the phrase "query result location" or "output location" (or at least, they theoretically could be -- it's what the fields are called in the web UI and CloudFormation for workgroups respectively) -- do they know what is being asked of them when we say `S3 staging dir`? I don't know what PyAthena is asking of us, so I doubt a partner would.
- One nice thing about `staging` is it avoids any confusion with the "ETL output dir" that the user previously set up
- I'm annoyed that PyAthena wants this at all. The code seems to only require workgroup or staging (which is perfectly consistent with "we'll grab the staging dir from the workgroup" -- dunno why it failed for me)

File: cumulus_library/cli.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dogversioning) [reply cumulus_library/cli.py:32]: i suppose i'm fine harmonizing on the AWS UI presentation name for user experience, even though from a technical perspective based on the underlying presto implementation i think schema is more correct. that's a Jeff beef though.

[the remark this answers is not in the document]
Comment (mikix) [reply cumulus_library/cli.py:32]: A split between the UI field name and the DB field name is a tale as old as time :smile:

[the remark this answers is not in the document]
Comment (mikix) [reply cumulus_library/cli.py:32]: I can change this PR to switch to `schema` earlier (i.e. no variables with `database`)

[the remark this answers is not in the document]
Comment (dogversioning) [reply cumulus_library/cli.py:32]: > A split between the UI field name and the DB field name is a tale as old as time 😄

sigh yes



> I can change this PR to switch to `schema` earlier (i.e. no variables with `database`)

works for me.

[the remark this answers is not in the document]
Comment (mikix) [reply cumulus_library/cli.py:32]: > sigh yes

I say it's a good thing! Tech folks are not end users, and we shouldn't force one to be the other.

File: cumulus_library/cli.py (hunk 1 above)
Comment (mikix) [reply cumulus_library/cli.py]: Renamed to `--s3-staging` and `CUMULUS_LIBRARY_S3_STAGING`, but I still worry that users will have no idea what that is.

Comment (mikix) [reply cumulus_library/cli.py]: OK we talked in person - and we both fixed our separate user-error errors. So now it looks like we can delete this option entirely. I've updated the PR

File: cumulus_library/cli.py [hunk 3]
```diff
@@ -338,5 +334,9 @@ def main(cli_args=None):
     return run_cli(args)
 
 
+def main_cli():  # called by the generated wrapper scripts
+    main()
```
Comment (mikix) [inline cumulus_library/cli.py:336]: This is so that the generated wrappers can avoid a return (which gets printed) and the unit tests can keep the return of the builder.

**Comment by dogversioning** [comment]
on line 142 in `study_parser.py`, we should change the text in the error to refer to databases instead of schemas.

**Comment by mikix** [comment]
> on line 142 in `study_parser.py`, we should change the text in the error to refer to databases instead of schemas.

Good catch! Fixed.
